src/System.Net.Sockets/tests/FunctionalTests/DisconnectTest.cs

```diff
@@ -67,16 +65,14 @@ public void Disconnect_Success()
 
                     args.RemoteEndPoint = server2.EndPoint;
 
-                    Assert.True(client.ConnectAsync(args));
-                    completed.WaitOne();
+                    if (client.ConnectAsync(args))
```

Comment:
Do we have any tests for `client.Disconnect(false)`?  If not, we should add one, and make sure that the Unix implementation respects the reuseSocket == false in a way similar to how Windows would (or, alternatively, throw a PlatformNotSupportedException for that restricted case if we can't support it, but it seems like that should be the easy case).